src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs

```diff
@@ -95,20 +133,396 @@ await signInManager.ValidateSecurityStampAsync(refreshTicket.Principal) is not T
             return TypedResults.SignIn(newPrincipal, authenticationScheme: IdentityConstants.BearerScheme);
         });
 
+        routeGroup.MapGet("/confirmEmail", async Task<Results<ContentHttpResult, UnauthorizedHttpResult>>
+            ([FromQuery] string userId, [FromQuery] string code, [FromQuery] string? changedEmail, [FromServices] IServiceProvider sp) =>
+        {
+            var userManager = sp.GetRequiredService<UserManager<TUser>>();
+            if (await userManager.FindByIdAsync(userId) is not { } user)
+            {
+                // We could respond with a 404 instead of a 401 like Identity UI, but that feels like unnecessary information.
+                return TypedResults.Unauthorized();
+            }
+
+            IdentityResult result;
+            try
+            {
+                code = Encoding.UTF8.GetString(WebEncoders.Base64UrlDecode(code));
+                if (string.IsNullOrEmpty(changedEmail))
+                {
+                    result = await userManager.ConfirmEmailAsync(user, code);
+                }
+                else
+                {
+                    result = await userManager.ChangeEmailAsync(user, changedEmail, code);
+                }
+            }
+            catch (FormatException)
+            {
+                return TypedResults.Unauthorized();
+            }
+
+            if (!result.Succeeded)
+            {
+                return TypedResults.Unauthorized();
+            }
+
+            return TypedResults.Text("Thank you for confirming your email.");
+        })
+        .Add(endpointBuilder =>
+        {
+            var finalPattern = ((RouteEndpointBuilder)endpointBuilder).RoutePattern.RawText;
+            confirmEmailEndpointName = $"{nameof(MapIdentityApi)}-{finalPattern}";
+            endpointBuilder.Metadata.Add(new EndpointNameMetadata(confirmEmailEndpointName));
+            endpointBuilder.Metadata.Add(new RouteNameMetadata(confirmEmailEndpointName));
+        });
```

Comment:
What is this for?

Reply:
It's to generate a unique name for /confirmEmail endpoint even if `MapIdentityApi` is called multiple times in different route groups. See the CanAddEndpointsToMultipleRouteGroupsForSameUserType and CanAddEndpointsToMultipleRouteGroupsForMultipleUsersTypes tests.